metricbeat/module/system/memory/memory.go

```diff
@@ -83,6 +88,21 @@ func (m *MetricSet) Fetch(r mb.ReporterV2) error {
 		},
 		"free": swapStat.Free,
 	}
+
+	if vmstat != nil {
+		swap["vmstat"] = common.MapStr{
+			// Swap in and swap out numbers
+			"pswpin":  vmstat.Pswpin,
+			"pswpout": vmstat.Pswpout,
+			// Hugepage swap stats
+			"thp_swpout":          vmstat.ThpSwpout,
+			"thp_swpout_fallback": vmstat.ThpSwpoutFallback,
+			//Swap readahead
+			"swap_ra":     vmstat.SwapRa,
+			"swap_ra_hit": vmstat.SwapRaHit,
+		}
+	}
+
```

Comment:
Instead of having `GetVMStat` and then add the fields to the event if the result is not nil, we could have something like `AddVMStat(e common.MapStr) err` that gets the vm stats and fills the available fields. This way we wouldn't need to care here in `fetch` about the different available metrics, and we would only need to add:
```
err := mem.AddVMStat(memory)
if err != nil {
  return errors.Wrap(err, "adding vm stat data")
}